src/content/docs/en/guides/server-side-rendering.mdx

```diff
@@ -222,7 +222,43 @@ See more details about [`Astro.cookies` and the `AstroCookie` type](/en/referenc
 
 ### `Response`
 
-You can also return a [Response](https://developer.mozilla.org/en-US/docs/Web/API/Response) from any page using on-demand rendering. 
+[`Astro.response`](/en/reference/api-reference/#astroresponse) is a standard [`ResponseInit`](https://developer.mozilla.org/en-US/docs/Web/API/Response/Response#options) object. It can be used to set the response status and headers.
+
+```astro title="src/pages/index.astro" {8-9}
+---
+import { getProduct } from '../api';
+
+const product = await getProduct(Astro.params.id);
+
+// No product found
+if (!product) {
+  Astro.response.status = 404;
+  Astro.response.statusText = 'Not found';
+}
+---
+<html>
+  <!-- Page here... -->
+</html>
+```
+
+Setting these values will change the response status and status text for the page.
```

Comment:
``````suggestion
[`Astro.response`](/en/reference/api-reference/#astroresponse) is a standard [`ResponseInit`](https://developer.mozilla.org/en-US/docs/Web/API/Response/Response#options) object. It can be used to set the response status and headers.

The example below sets a response status and status text for a product listing page when the product does not exist:

```astro title="src/pages/my-product.astro" {8-9}
---
import { getProduct } from '../api';

const product = await getProduct(Astro.params.id);

// No product found
if (!product) {
  Astro.response.status = 404;
  Astro.response.statusText = 'Not found';
}
---
<html>
  <!-- Page here... -->
</html>
```
``````
Great example here! Fix this for accuracy, but we typically like to introduce code snippets with the explanation of what's happening first to prime them for reading the code. This keeps the reader flowing down the page, instead of seeing code, then reading an explanation, then scrolling back up to check what the explanation referred to.

I also presented a suggestion for making the text context-specific for the example, but feel free to use something that similarly explains the code in context! (Including the file name of the snippet!)

Reply:
That is better, thanks!